### PR TITLE
Clean 'meson dist' command

### PR DESCRIPTION
**Pull request by michaelbadcrumble**
Changed Python2 %s syntax to Python3 format and added comments.

### CODE REVIEW

File: mesonbuild/mdist.py [hunk 1]
```diff
@@ -33,6 +33,10 @@
                      'zip': '.zip'}
 
 def add_arguments(parser):
+    '''
+      here we are adding arguments that can be used with
```
Comment (nirbheek) [inline mesonbuild/mdist.py]: You don't need to start each comment with "here we are", because it's redundant.

File: mesonbuild/mdist.py [hunk 2]
```diff
@@ -233,7 +285,11 @@ def determine_archives_to_generate(options):
         sys.exit('No archive types specified.')
     return result
 
+
 def run(options):
+    '''
+    this is a runner function for 'meson dist'
```
Comment (nirbheek) [inline mesonbuild/mdist.py]: You don't need to say "this is a", just say `runner function for 'meson dist'`. Same for all the rest.

Comment (michaelbadcrumble) [reply mesonbuild/mdist.py]: Sorry, my brain was on autopilot, will fix this.

File: mesonbuild/mdist.py (hunk 1 above)
Comment (michaelbadcrumble) [reply mesonbuild/mdist.py]: Problem fixed.

**Comment by jpakkane** [comment]
Thanks, but I'm going to have to be a slight contrarian here w.r.t. the comments. See for example this:

```
def is_hg(src_root):
    '''
    Check to see if its a 'hg' project.
    '''
```

This comment provides no extra information, it is simply saying the same thing as the function name. Thus it is visual noise that should be removed. If one wants to clarify that this is a "project" then the better approach would be to rename the function to `is_hg_project` instead. (I don't think it is necessary in this case.)

The same goes for the other added comment blocks.

**Comment by michaelbadcrumble** [comment]
Needless comments removed.

**Comment by nirbheek** [comment]
Did you close this PR on purpose?

**Comment by michaelbadcrumble** [comment]
I wasn’t particularly happy with that PR.  So I created #6725 since I originally wanted to cherry-pick Python2 syntax.  Also, note-to-self no PR should be created late at night or without having a cup coffee first.

**Pull request closed without merging**